src/Servers/Kestrel/test/InMemory.FunctionalTests/MaxRequestBodySizeTests.cs

```diff
@@ -60,6 +60,89 @@ public async Task RejectsRequestWithContentLengthHeaderExceedingGlobalLimit()
+    {
+        var maxRequestBodySize = 10;
+        await using (var server = new TestServer(async context =>
+        {
+#pragma warning disable CS0618 // Type or member is obsolete
+            BadHttpRequestException requestRejectedEx = await Assert.ThrowsAsync<BadHttpRequestException>(async () =>
+#pragma warning restore CS0618 // Type or member is obsolete
+            {
+                using (var stream = new StreamReader(context.Request.Body))
+                {
+                    string body = await stream.ReadToEndAsync();
+                }
+            });
+            context.Response.StatusCode = 413;
+            await context.Response.WriteAsync("");
+            throw requestRejectedEx;
+        },
+        new TestServiceContext(LoggerFactory) { ServerOptions = new KestrelServerOptions { Limits = { MaxRequestBodySize = maxRequestBodySize } } }))
+        {
+            using var connection = server.CreateConnection();
+            await connection.Send(
+                "POST / HTTP/1.1",
+                "Host:",
+                "Content-Length: " + (maxRequestBodySize + 1),
+                "",
+                new string('a', maxRequestBodySize + 1));
+            await connection.ReceiveEnd(
+                "HTTP/1.1 413 Payload Too Large",
+                "Connection: close",
+                $"Date: {server.Context.DateHeaderValue}",
+                "Server: Kestrel",
+                "Transfer-Encoding: chunked",
+                "",
+                "");
+        }
+    }
+
+    [Fact]
+    public async Task RejectsRequestWithChunckedBodySizeExceedingPerRequestLimitAndExceptionWasCaughtByApplication()
+    {
+        var maxRequestBodySize = 10;
+        var chunkedPayload = $"5;random chunk extension\r\nHello\r\n6\r\n World\r\n0\r\n";
+        await using (var server = new TestServer(async context =>
+        {
+#pragma warning disable CS0618 // Type or member is obsolete
+            BadHttpRequestException requestRejectedEx = await Assert.ThrowsAsync<BadHttpRequestException>(async () =>
+#pragma warning restore CS0618 // Type or member is obsolete
+            {
+                using (var stream = new StreamReader(context.Request.Body))
+                {
+                    string body = await stream.ReadToEndAsync();
+                }
+            });
+            context.Response.StatusCode = 413;
+            await context.Response.WriteAsync("");
+            throw requestRejectedEx;
+        },
+        new TestServiceContext(LoggerFactory) { ServerOptions = new KestrelServerOptions { Limits = { MaxRequestBodySize = maxRequestBodySize } } }))
+        {
+            using var connection = server.CreateConnection();
+            await connection.Send(
+                "POST / HTTP/1.1",
+                "Host:",
+                "Transfer-Encoding: chunked",
+                "",
+                chunkedPayload + "POST / HTTP/1.1",
```

Comment:
Is this second request supposed to be here?

Reply:
No, sorry, I'll remove this. This method is only to test a chunked request, not necessarily 2 requests in the same stream.